### PR TITLE
KYLIN-4162 After the paragraph is dropped, continue to build the subsequent segment and provide REST API to resubmit the segment

### PR DESCRIPTION
**Pull request by zhangayqian**

### CODE REVIEW

File: server-base/src/main/java/org/apache/kylin/rest/controller/CubeController.java [hunk 1]
```diff
@@ -855,6 +863,38 @@ public CubeRequest updateCubeDesc(@RequestBody CubeRequest cubeRequest) throws J
...
+            CubingJob cubingJob = (CubingJob) executable;
+            newSeg.setLastBuildJobID(cubingJob.getId());
+
+            SegmentBuildState.BuildState state = new SegmentBuildState.BuildState();
+            state.setBuildStartTime(System.currentTimeMillis());
+            state.setState(SegmentBuildState.BuildState.State.BUILDING);
+            state.setJobId(cubingJob.getId());
+            streamMetadataStore.updateSegmentBuildState(cubeName, segmentName, state);
+        } catch (Exception e) {
+            logger.error("streaming job submit fail, cubeName:" + cubeName + " segment:" + segmentName, e);
```
Comment (nichunen) [inline server-base/src/main/java/org/apache/kylin/rest/controller/CubeController.java]: Do not eat up the exception, throw it.

File: server-base/src/main/java/org/apache/kylin/rest/controller/CubeController.java [hunk 2]
```diff
@@ -855,6 +863,38 @@ public CubeRequest updateCubeDesc(@RequestBody CubeRequest cubeRequest) throws J
 
     }
 
+    /**
+     * resubmit streaming segment
+     *
+     * @throws IOException
+     */
+    @RequestMapping(value = "/{cubeName}/segs/{segmentName}/resubmit", method = {RequestMethod.PUT}, produces = {"application/json"})
```
Comment (nichunen) [inline server-base/src/main/java/org/apache/kylin/rest/controller/CubeController.java]: How about adding this api as an action item for a streaming cube?

File: server-base/src/main/java/org/apache/kylin/rest/controller/CubeController.java (hunk 1 above)
Comment (zhangayqian) [reply server-base/src/main/java/org/apache/kylin/rest/controller/CubeController.java]: Ok, I got it.

File: server-base/src/main/java/org/apache/kylin/rest/controller/CubeController.java (hunk 2 above)
Comment (zhangayqian) [reply server-base/src/main/java/org/apache/kylin/rest/controller/CubeController.java]: May I need to add a 'resubmit segment' action item to the streaming cube page?

File: server-base/src/main/java/org/apache/kylin/rest/controller/CubeController.java [hunk 3]
```diff
@@ -855,6 +863,40 @@ public CubeRequest updateCubeDesc(@RequestBody CubeRequest cubeRequest) throws J
 
     }
 
+    /**
+     * resubmit streaming segment
+     *
+     * @throws IOException
+     */
+    @RequestMapping(value = "/{cubeName}/segs/{segmentName}/resubmit", method = { RequestMethod.PUT }, produces = {
```
Comment (nichunen) [inline server-base/src/main/java/org/apache/kylin/rest/controller/CubeController.java]: Why not put the api in JobController?

**Pull request merged**